kivy_garden/mapview/utils.py

```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
+
+    # map_width /= 5
+    # map_height /= 5
+
+    # Constants for zoom calculation
+    TILE_SIZE = 256
+
+    # Function to convert latitude to pixel value
+    def lat_to_pixel(lat, zoom):
+        return (
+            TILE_SIZE
+            * (1 - log(tan(radians(lat)) + 1 / cos(radians(lat))) / pi)
+            / 2 * (2 ** zoom)
+        )
+
+    # Function to convert longitude to pixel value
+    def lon_to_pixel(lon, zoom):
+        return TILE_SIZE * (lon + 180) / 360 * (2 ** zoom)
+
+    # Determine the best zoom level
+    zoom = 1
+    for z in range(1, 21):  # Assuming a max zoom level of 20
+        lat_pixel_range = lat_to_pixel(max_lat, z) - lat_to_pixel(min_lat, z)
+        lon_pixel_range = lon_to_pixel(max_lon, z) - lon_to_pixel(min_lon, z)
+
+        if lat_pixel_range < map_height and lon_pixel_range < map_width:
+            zoom = z
+        else:
+            break
+
+    return zoom
+
+
+def update_map_view(
+        map_width,
+        map_height,
+        lat1,
+        lon1,
+        lat2,
+        lon2,
+        mapview=None,
+        polyline_layer=None,
+        max_zoom=16
+
+):
+    """
+    Updates the MapView to ensure that two specified
+    locations are both visible on the screen, centering the
+    view between the two locations and adjusting the zoom level
+    accordingly.
+
+    This function calculates the optimal center point and zoom
+    level for the MapView to display both `(lat1, lon1)` and `(lat2, lon2)`.
+    It ensures that the map is centered between these two points and
+    adjusts the zoom level so that both locations remain visible within
+    the given map dimensions.
+
+    The function performs the following steps:
+    1. Calculates the geographic center between
+        `(lat1, lon1)` and `(lat2, lon2)`.
+    2. Determines the appropriate zoom level to fit both locations within
+        the specified `map_width` and `map_height`.
+    3. Further adjusts the zoom level based on the distance between the two
+        locations using the Haversine formula.
+    4. Centers the map on the calculated center point.
+    5. Sets the zoom level to the average of the calculated zoom levels, with a
+        maximum zoom level of 16.
+    6. Updates the coordinates for the polyline layer to draw a line between
+        `(lat1, lon1)` and `(lat2, lon2)`.
+
+    """
+    coordinates = [(lat1, lon1), (lat2, lon2)]
+    center_lat, center_lon = get_bounding_box_center(coordinates)
+    z1 = get_fit_zoom_level(
+        coordinates,
+        map_width,
+        map_height
+    )
+    z2 = get_zoom_for_radius(haversine(lon1, lat1, lon2, lat2))
+    zoom_level = int((z1 + z2) / 2)
+    if mapview:
+        mapview.center_on(center_lat, center_lon)
+        mapview.zoom = min(zoom_level, max_zoom)
+    if polyline_layer:
+        polyline_layer.coordinates = coordinates
+    return (center_lat, center_lon), zoom_level
+
+
+def findpoints(lat, lon, radius):
```

Comment:
Please rename the function to something that match the intent. Something like `generate_circle_points()`.
Add the precision/count (currently N) as part of the parameters.